### PR TITLE
Clear chat on error and explain

### PR DESCRIPTION
**Pull request by aarondr77**
# Description

Clears the chat history when the user explains a code block or debugs an error. Clearing that chat history periodically leads to bette results. Resolves https://github.com/mito-ds/mito/issues/1359

# Testing

See the new test + try it out

# Documentation

None needed.

### CODE REVIEW

File: mito-ai/src/Extensions/AiChat/ChatMessage/ChatMessage.tsx [hunk 1]
```diff
@@ -85,14 +85,15 @@ const ChatMessage: React.FC<IChatMessageProps> = ({
             { "message-user": message.role === 'user' },
             { 'message-assistant': message.role === 'assistant' },
         )}>
-            {messageContentParts.map(messagePart => {
+            {messageContentParts.map((messagePart, index) => {
+                console.log('messagePart', messagePart)
```
Comment (aarondr77) [inline mito-ai/src/Extensions/AiChat/ChatMessage/ChatMessage.tsx:89]:
```suggestion

```

**Review by aarondr77 — COMMENTED** [review]
small cleanup

**Review by ngafar — APPROVED** [review]

**Comment by ngafar** [comment]
Looks good. Both the "explain" and "debug" workflows work as expected. Merging after the CI/CD tests are completed.

**Comment by vibhatha** [comment]
@aarondr77 I was late to review this PR. I think we have a good start here.

**Pull request merged**